### ISSUE TITLE
carry over midi detune to detune patterns

### ISSUE THREAD

**Issue by RiedleroD** [LMMS/lmms#6299]
### Enhancement Summary

When importing midi, it would be nice to convert the pitch automation to detune patterns, similar to how it's done in #6297.

#### Justification

Whenever I import midi files, and there's some automation, I like to declutter the project by removing unneded automation duplicates and carry over pitchbends into the note editor - the latter is pretty time-consuming and also not very precise when done by hand, so It'd be very handy if it did that automatically.

**Comment by RiedleroD** [comment]
Possibly, this could be done in a popup post-import, asking the user if they want to carry the detune over to the detune patterns. Would be very nice to combine with other cleanup stuff …for which I will open feature requests eventually.

**Comment by Monospace-V** [comment]
Potential problem: if detune is carried over multiple notes. Enhancement also likely to result in less accurate detune than expected in some cases. However, since this is rare, I cannot comment. It is likely, in most cases, that carrying over detune to notes will work properly.
Other problem: individual notes of SF2 player and a couple of other plugins cannot be pitch bent, and trying to pitch bend it in-piano roll only transposes the entire note. The only way to pitch bend on these instruments is with the pitch knob.
But importing a MIDI by-default places everything on SF2 players. That would need to be fixed first.

**Comment by RiedleroD** [comment]
> is pitch bend commonly used for single notes?

Yeah, I've pretty much never seen midi files with multiple notes being bent; it's used a lot to compensate for not having a detune pattern in the midi standard. In my use-cases, transcriptions of Gameboy tracks use this to account for pitchbends in the first channel of the Gameboy. The Gameboy has 4 monophonic channels, fyi.

> An option to carry over detune at any point would be more useful.

Absolutely, great idea! Perhaps as a right-click option on an automation pattern? Something like 'copy pitchbends to notes'.

> Individual notes of SF2 player and a couple of other plugins cannot be pitch bent.

I know - though I only use the sf2 player for percussion, this is something that, imo, should be fixed regardless of this PR before 1.3.0 is released.

**Comment by Monospace-V** [comment]
An ability to do the opposite would be pretty useful, too. Given https://github.com/LMMS/lmms/pull/6297 now forces pitch bend messages to translate into piano roll, the ability to convert backwards from PR pitch bend to MIDI automation would be convenient for the same reasons MIDI to Piano Roll _could_ pose a problem.
